nomos-da/kzgrs/src/common.rs

```diff
@@ -0,0 +1,176 @@
+            .collect(),
+        domain,
+    )
+}
+
+/// Transform chunks of bytes (of size `CHUNK_SIZE`) into `Fr` which are considered evaluations of a
+/// polynomial. Then use FFT to transform that polynomial into coefficient form.
+/// `CHUNK_SIZE` needs to be 31 (bytes) or less, otherwise it cannot be encoded.
+/// The input data need to be padded, so it fits in a len modulus of `CHUNK_SIZE`.
+/// Returns the polynomial in evaluation form and in coefficient form
+pub fn bytes_to_polynomial<const CHUNK_SIZE: usize>(
+    data: &[u8],
+    domain: GeneralEvaluationDomain<Fr>,
+) -> Result<(Evaluations<Fr>, DensePolynomial<Fr>), KzgRsError> {
+    if CHUNK_SIZE > BYTES_PER_FIELD_ELEMENT {
+        return Err(KzgRsError::ChunkSizeTooBig(CHUNK_SIZE));
+    }
+    if data.len() % CHUNK_SIZE != 0 {
+        return Err(KzgRsError::UnpaddedDataError {
+            expected_modulus: CHUNK_SIZE,
+            current_size: data.len(),
+        });
+    }
+    Ok(bytes_to_polynomial_unchecked::<CHUNK_SIZE>(data, domain))
+}
+
+/// Transform chunks of bytes (of size `CHUNK_SIZE`) into `Fr` which are considered evaluations of a
+/// polynomial. Then use FFT to transform that polynomial into coefficient form.
+/// No extra checks are done for the caller.
+/// Caller need to ensure that `CHUNK_SIZE` is not bigger than the underlying `Fr` element can be
+/// decoded from.
+pub fn bytes_to_polynomial_unchecked<const CHUNK_SIZE: usize>(
+    data: &[u8],
+    domain: GeneralEvaluationDomain<Fr>,
+) -> (Evaluations<Fr>, DensePolynomial<Fr>) {
+    let evals = bytes_to_evaluations::<CHUNK_SIZE>(data, domain);
+    let coefficients = evals.interpolate_by_ref();
+    (evals, coefficients)
+}
+
+/// Transform arbitrary bytes into a field element
+/// This transformation is bounds unchecked, it's up to the caller to know if
+/// data fits within the bls modulus.
+/// Data len cannot be higher than `BYTES_PER_FIELD_ELEMENT`
+pub fn field_element_from_bytes_le(b: &[u8]) -> FieldElement {
+    assert!(b.len() <= BYTES_PER_FIELD_ELEMENT);
+    FieldElement::from(BigUint::from_bytes_le(b))
+}
+
+#[cfg(test)]
+mod test {
+    use super::{bytes_to_evaluations, bytes_to_polynomial, KzgRsError};
+    use ark_bls12_381::fr::Fr;
+    use ark_ff::{BigInteger, PrimeField};
+    use ark_poly::{EvaluationDomain, GeneralEvaluationDomain, Polynomial};
+    use once_cell::sync::Lazy;
+    use rand::{thread_rng, Fill};
+
+    const CHUNK_SIZE: usize = 31;
+    static DOMAIN: Lazy<GeneralEvaluationDomain<Fr>> =
+        Lazy::new(|| GeneralEvaluationDomain::new(128).unwrap());
+    #[test]
+    fn encode_random_polynomial() {
+        const N: usize = 100;
+        let mut bytes: [u8; CHUNK_SIZE * N] = [0; CHUNK_SIZE * N];
+        let mut rng = thread_rng();
+        bytes.try_fill(&mut rng).unwrap();
+        let evals = bytes_to_evaluations::<31>(&bytes, *DOMAIN);
+        let (_, poly) = bytes_to_polynomial::<31>(&bytes, *DOMAIN).unwrap();
+        for i in 0..100 {
```

Comment:
```suggestion
        for i in 0..N {
```
Can we probably use N?